File: ScoreFrame/ScoreFrame/Models/TimerSegment.swift
import Foundation

struct TimerSegment: Codable, Identifiable, Equatable {
    var id: UUID = UUID()
    var periodLabel: String?
    var segmentStartTime: TimeInterval?  // 区切り開始（ピリオド切替・タイマー初期値表示）
    var timerStartTime: TimeInterval?    // キックオフ（タイマー計測開始）
    var timerStopTime: TimeInterval?
    var timerStartOffset: TimeInterval?

    // タイマー表示オプション
    var showPlusPrefix: Bool = false     // "+MM:SS" 形式で表示
    var timerColorHex: String? = nil     // タイマー文字色 ("#RRGGBB")

    /// ピリオド切替・タイマー表示の起点。segmentStartTime が未設定なら timerStartTime を使う
    var effectiveStartTime: TimeInterval? {
        segmentStartTime ?? timerStartTime
    }
}

File: ScoreFrame-Android/app/src/main/java/com/scoreframe/model/TimerSegment.kt
package com.scoreframe.model

import kotlinx.serialization.Serializable

@Serializable
data class TimerSegment(
    val id: String = java.util.UUID.randomUUID().toString(),
    var periodLabel: String? = null,
    var segmentStartTime: Double? = null,
    var timerStartTime: Double? = null,
    var timerStopTime: Double? = null,
    var timerStartOffset: Double? = null,
    var showPlusPrefix: Boolean = false,
    var timerColorHex: String? = null
) {
    val effectiveStartTime: Double?
        get() = segmentStartTime ?: timerStartTime
}
